contracts/src/rollup/BOLDUpgradeAction.sol

```diff
@@ -155,6 +155,19 @@ contract RollupReader is IOldRollup {
     }
 }
 
+/// @notice Stores an array specified during construction.
+///         Since the BOLDUpgradeAction is not allowed to have storage, 
```

Comment:
geometric mini stake is pretty useless when there are too many level (it will be very small at the end) so I think we can assume we are only going to have a limited amount of level before we use a fixed amount

with this assumption in mind, you can define the stake array as a static array e.g. miniStakes[5] so that it can be store as immutable, in the mini-staking logic we can have all level beyond idx 5 use the value of idx 5

Reply:
hmm well since it is a one-time thing and we already have another 2 helper contract, let’s stick to the current design